modules/savebuff.cpp

```diff
@@ -94,115 +95,119 @@ class CSaveBuff : public CModule
-				CString sLine(*it);
-				sLine.Trim();
-				if (sLine[0] == '@' && it+1 != vsLines.end())
-				{
-					CString sTimestamp = sLine.Token(0);
-					sTimestamp.TrimLeft("@");
-					timeval ts;
-					ts.tv_sec = sTimestamp.Token(0, false, ",").ToLongLong();
-					ts.tv_usec = sTimestamp.Token(1, false, ",").ToLong();
-
-					CString sFormat = sLine.Token(1, true);
-
-					CString sText(*++it);
-					sText.Trim();
-
-					pTarget->AddBuffer(sFormat, sText, &ts);
-				} else
-				{
-					// Old format, escape the line and use as is.
-					pTarget->AddBuffer(_NAMEDFMT(sLine));
-				}
+		for (it = vsLines.begin(); it != vsLines.end(); ++it) {
+			CString sLine(*it);
+			sLine.Trim();
+			if (sLine[0] == '@' && it+1 != vsLines.end())
+			{
+				CString sTimestamp = sLine.Token(0);
+				sTimestamp.TrimLeft("@");
+				timeval ts;
+				ts.tv_sec = sTimestamp.Token(0, false, ",").ToLongLong();
+				ts.tv_usec = sTimestamp.Token(1, false, ",").ToLong();
+
+				CString sFormat = sLine.Token(1, true);
+
+				CString sText(*++it);
+				sText.Trim();
+
+				pTarget->AddBuffer(sFormat, sText, &ts);
+			} else
+			{
+				// Old format, escape the line and use as is.
+				pTarget->AddBuffer(_NAMEDFMT(sLine));
 			}
-		} else
-		{
-			m_sPassword = "";
-			CUtils::PrintError("[" + GetModName() + ".so] Failed to Decrypt [" + pTarget->GetName() + "]");
-			return(false);
 		}
+	}
+
+	void SaveBufferToDisk(const CBuffer& Buffer, const CString& sName, const CString& sToken)
+	{
+		CString sPath = GetPath(sName);
+		CFile File(sPath);
+
+		CString sContent = sToken + sName + "\n";
+
+		size_t uSize = Buffer.Size();
+		for (unsigned int uIdx = 0; uIdx < uSize; uIdx++) {
+			const CBufLine& Line = Buffer.GetBufLine(uIdx);
+			timeval ts = Line.GetTime();
+			sContent +=
+				"@" + CString(ts.tv_sec) + "," + CString(ts.tv_usec) + " " +
+				Line.GetFormat() + "\n" +
+				Line.GetText() + "\n";
+		}
+
+		CBlowfish c(m_sPassword, BF_ENCRYPT);
+		sContent = c.Crypt(sContent);
 
-		return(true);
+		if (File.Open(O_WRONLY | O_CREAT | O_TRUNC, 0600)) {
+			File.Chmod(0600);
+			File.Write(sContent);
+		}
+		File.Close();
 	}
 
-	void SaveBufferToDisk()
+	void SaveBuffersToDisk()
 	{
 		if (!m_sPassword.empty())
 		{
-			const vector<CChan *>& vChans = GetNetwork()->GetChans();
-			for (u_int a = 0; a < vChans.size(); a++)
-			{
-				CString sPath = GetPath(vChans[a]->GetName());
-				CFile File(sPath);
-
-				const CBuffer& Buffer = vChans[a]->GetBuffer();
-				CString sLine;
-
-				CString sFile = CRYPT_VERIFICATION_TOKEN;
-
-				size_t uSize = Buffer.Size();
-				for (unsigned int uIdx = 0; uIdx < uSize; uIdx++) {
-					const CBufLine& Line = Buffer.GetBufLine(uIdx);
-					timeval ts = Line.GetTime();
-					sFile +=
-						"@" + CString(ts.tv_sec) + "," + CString(ts.tv_usec) + " " +
-						Line.GetFormat() + "\n" +
-						Line.GetText() + "\n";
-				}
+			// cleanup auto-cleared queries
+			CDir::Delete("*", GetSavePath());
```

Comment:
What if ZNC crashes right after this? I think it's better to delete unused files after writing new files.